File: ActivityStateManagementApp/app/src/main/java/com/pathtech/activitystatemanagementapp/MainActivity.java
package com.pathtech.activitystatemanagementapp;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import java.io.Serializable;
import java.util.Date;

public class MainActivity extends AppCompatActivity {

    public static final String KEY_TXT_VW = "txtVw";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mt("onCreate");

        /*if (savedInstanceState != null) {
            State state = (State) savedInstanceState.getSerializable(KEY_TXT_VW);
            if (state != null)
                ((TextView) findViewById(R.id.txtVw)).setText(state.txtVwData);
        }*/
    }

    public void okayClicked(View view) {
        ((TextView) findViewById(R.id.txtVw)).setText(new Date().toString());
    }

    @Override
    protected void onDestroy() {
        mt("onDestroy");
        super.onDestroy();
    }

    private void mt(String msg) {

        Toast.makeText(this, msg, Toast.LENGTH_SHORT).show();
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);

        State state = new State();
        state.txtVwData = ((TextView) findViewById(R.id.txtVw)).getText().toString();

        outState.putSerializable(KEY_TXT_VW, state);
    }

    class State implements Serializable {
        public String txtVwData;
    }

    @Override
    protected void onRestoreInstanceState(Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);

        if (savedInstanceState != null) {
            State state = (State) savedInstanceState.getSerializable(KEY_TXT_VW);
            if (state != null)
                ((TextView) findViewById(R.id.txtVw)).setText(state.txtVwData);
        }
    }
}

File: PaymentTest/PaymentTestApp/app/src/main/java/com/pathtech/paymenttestapp/MainActivity.kt
package com.pathtech.paymenttestapp

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.sharath.upi.payment.UpiPayment
import com.sharath.upi.payment.model.PaymentDetail
import com.sharath.upi.payment.model.TransactionDetails
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.AnkoLogger
import org.jetbrains.anko.info

class MainActivity : AppCompatActivity(), AnkoLogger {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        id_pay_using_upi_app.setOnClickListener {
            startUpiPayment()
        }
    }


    private fun startUpiPayment(){
        val payment = PaymentDetail(
            vpa="9538555123@ybl",
            name = "Sharathkumar K G",
            payeeMerchantCode = "",
            //txnId = "",
            txnRefId = "",
            description = "description",
            amount = "1.00")


        UpiPayment(this)
            .setPaymentDetail(payment)
            .setUpiApps(UpiPayment.UPI_APPS)
            .setCallBackListener(object : UpiPayment.OnUpiPaymentListener{
                override fun onSubmitted(data: TransactionDetails) {
                    info { "transaction pending: $data" }
                    Toast.makeText(this@MainActivity,"transaction pending: $data",Toast.LENGTH_LONG).show()
                }
                override fun onSuccess(data: TransactionDetails) {
                    info { "transaction success: $data" }
                    Toast.makeText(this@MainActivity,"transaction success: $data",Toast.LENGTH_LONG).show()
                }
                override fun onError(message: String) {
                    info { "transaction failed: $message" }
                    Toast.makeText(this@MainActivity,"transaction failed: $message",Toast.LENGTH_LONG).show()
                }
            }).pay()


        val existingApps = UpiPayment.getExistingUpiApps(this)
        info { "existing app: $existingApps" }
    }
}
